tests/nlu/classifiers/test_diet_classifier.py

```diff
@@ -701,68 +701,139 @@ def test_removing_label_sparse_feature_sizes(
     assert sparse_feature_sizes == final_sparse_feature_sizes
 
 
-@pytest.mark.timeout(120, func_only=True)
-async def test_adjusting_layers_incremental_training(component_builder, tmpdir):
+@pytest.mark.timeout(120)
+async def test_adjusting_layers_incremental_training(
+    component_builder: ComponentBuilder, tmpdir: Path
+):
     """Tests adjusting sparse layers to increased sparse feature sizes
-       during incremental training."""
+       during incremental training.
+
+       Tests if the layers were replaced by checking their sizes.
+       Checking if they were replaces correctly is also important
+       and is done in `test_replace_dense_for_sparse_layers`
+       in `test_rasa_layers.py`.
+       """
     pipeline = [
         {"name": "WhitespaceTokenizer"},
+        {"name": "LexicalSyntacticFeaturizer"},
+        {"name": "RegexFeaturizer"},
         {"name": "CountVectorsFeaturizer"},
-        {"name": "DIETClassifier", LOSS_TYPE: "margin", EPOCHS: 1},
+        {
+            "name": "CountVectorsFeaturizer",
+            "analyzer": "char_wb",
+            "min_ngram": 1,
+            "max_ngram": 4,
+        },
+        {"name": "DIETClassifier", EPOCHS: 1},
+        {"name": "ResponseSelector", EPOCHS: 1},
```

Comment:
The test for `ResponseSelector` in the pipeline should be in its own file `tests/nlu/selectors/test_response_selector.py`